submission/distributor_test.go

```diff
@@ -222,3 +271,126 @@ func TestNewDistributorRootPools(t *testing.T) {
 		})
 	}
 }
+
+func pemFileToDERChain(t *testing.T, filename string) [][]byte {
+	t.Helper()
+	rawChain, err := x509util.ReadPossiblePEMFile(filename, "CERTIFICATE")
+	if err != nil {
+		t.Fatalf("failed to load testdata: %v", err)
+	}
+	return rawChain
+}
+
+func getSCTMap(l []*AssignedSCT) map[string]*AssignedSCT {
+	m := map[string]*AssignedSCT{}
+	for _, asct := range l {
+		m[asct.LogURL] = asct
+	}
+	return m
+}
+
+// Stub CT policy to run tests.
+type stubCTPolicy struct {
+	baseNum int
+}
+
+// Builds simplistic policy requiring n SCTs from any Logs for each cert.
+func buildStubCTPolicy(n int) stubCTPolicy {
+	return stubCTPolicy{baseNum: n}
+}
+
+func (stubP stubCTPolicy) LogsByGroup(cert *x509.Certificate, approved *loglist.LogList) (ctpolicy.LogPolicyData, error) {
+	baseGroup, err := ctpolicy.BaseGroupFor(approved, stubP.baseNum)
+	groups := ctpolicy.LogPolicyData{baseGroup.Name: &baseGroup}
+	return groups, err
+}
+
+func TestDistributorAddPreChain(t *testing.T) {
+	testCases := []struct {
+		name     string
+		ll       *loglist.LogList
+		plc      ctpolicy.CTPolicy
+		rawChain [][]byte
+		scts     []*AssignedSCT
+		wantErr  bool
+	}{
+		{
+			name:     "MalformedChainRequest",
+			ll:       sampleValidLogList(t),
+			plc:      ctpolicy.ChromeCTPolicy{},
+			rawChain: pemFileToDERChain(t, "../trillian/testdata/subleaf.misordered.chain"),
+			scts:     nil,
+			wantErr:  true,
+		},
+		{
+			name:     "CallBeforeInit",
+			ll:       sampleValidLogList(t),
+			plc:      ctpolicy.ChromeCTPolicy{},
+			rawChain: nil,
+			scts:     nil,
+			wantErr:  true,
+		},
+		{
+			name:     "FullChainApplePolicy",
```

Comment:
It's most helpful if the test name indicates what behaviour you're testing. In this case, your name describes the test data, rather than the behaviour. 
```suggestion
			name:     "InsufficientSCTsForPolicy",
```

Reply:
renamed.